.github/workflows/codeql2.yml

```diff
@@ -0,0 +1,92 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL Advanced"
```

Comment:
Could: It would be better to add this in a separate PR linking to an issue specifically for adding CodeQL, otherwise we're mixing unrelated things.

Reply:
if i18 was tangled I'd make that from the start but it's an early exploratory phase still at i18 so I accept the things as in flux.


There are no well defined acceptance criteria yet, this is a proof of concept that we can scan with the hardware and software as they are at the moment. 

undulators turned out not to be relevant to this issue and PR.

Reply:
extracted here
https://github.com/DiamondLightSource/i18-bluesky/issues/25